addons/website_sale_stock/static/src/js/combo_configurator_dialog/combo_configurator_dialog.js

```diff
@@ -0,0 +1,47 @@
+import { patch } from '@web/core/utils/patch';
+import { useSubEnv } from '@odoo/owl';
+import {
+    ComboConfiguratorDialog
+} from '@sale/js/combo_configurator_dialog/combo_configurator_dialog';
+
+patch(ComboConfiguratorDialog.prototype, {
+    setup() {
+        super.setup(...arguments);
+
+        useSubEnv({
+            isProductQuantityAllowed: this.isProductQuantityAllowed.bind(this),
+        });
+    },
+
+    async selectComboItem(comboId, comboItem) {
+        if (!this.isProductQuantityAllowed(comboItem.product, this.state.quantity)) {
+            return;
+        }
+        super.selectComboItem(...arguments);
+    },
+
+    /**
+     * Check whether the provided combo quantity can be added to the cart.
+     *
+     * @param {Number} quantity The quantity to check.
+     * @return {Boolean} Whether the combo quantity can be added to the cart.
+     */
+    isComboQuantityAllowed(quantity) {
+        return this._selectedComboItems.every(
+            comboItem => this.isProductQuantityAllowed(comboItem.product, quantity)
+        );
+    },
+
+    /**
+     * Check whether the provided product quantity can be added to the cart.
+     *
+     * @param {ProductProduct} product The provided product.
+     * @param {Number} quantity The quantity to check.
+     * @return {Boolean} Whether the product quantity can be added to the cart.
+     */
+    isProductQuantityAllowed(product, quantity) {
+        return !product.is_storable ||
+            product.allow_out_of_stock_order ||
+            product.free_qty >= quantity;
```

Comment:
Yeah, and I could also only send `free_qty` and not the other 2 (since it would only be populated iff `product.is_storable && !product.allow_out_of_stock_order`.

The reason I didn't do this is because we may want to display the available quantity in a future version (similarly to the product page). I'm not sure how likely that is though, so I'm fine with simplifying this code for now.

Reply:
Well, it can be costly in perf, and won't be shown in 18, so I believe we could avoid loading it when unnecessary.